src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant.Tests/IntegrationTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Net;
+using Duende.IdentityServer.Models;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace GSoft.Extensions.Http.Authentication.ClientCredentialsGrant.Tests;
+
+public class IntegrationTests
+{
+    private readonly ITestOutputHelper _testOutputHelper;
+
+    public IntegrationTests(ITestOutputHelper testOutputHelper)
+    {
+        this._testOutputHelper = testOutputHelper;
+    }
+
+    [Fact]
+    public async Task Real_Client_Server_Communication()
```

Comment:
This test boostrap a real ASP.NET Core app/host running in-memory.

- We add Duende's IdentityServer as a real identity provider to issue and validate JWTs.
- We also add some protected endpoints as the protected resource we're trying to communicate with.
- Finally, we create a authenticated HttpClient to communicate with the protected resource.